compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_21/service.bal

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+// Positive Cases
+
+type Album readonly & record {|
+    string id;
+    string title;
+    string artist;
+    decimal price;
+|};
+
+// albums table to seed record album data.
+table<Album> key(id) albums = table [
+        {id: "1", title: "Blue Train", artist: "John Coltrane", price: 56.99},
+        {id: "2", title: "Jeru", artist: "Gerry Mulligan", price: 17.99},
+        {id: "3", title: "Sarah Vaughan and Clifford Brown", artist: "Sarah Vaughan", price: 39.99}
+    ];
+
+service / on new http:Listener(8080) {
+    resource function post album(@http:Payload Album album) returns Album {
```

Comment:
Shall we add a test case for `album[]` too?

Reply:
Fixed with [1bee102577c66aac843862e5cea86169d0bffc4f](https://github.com/ballerina-platform/module-ballerina-http/pull/778/commits/1bee102577c66aac843862e5cea86169d0bffc4f)